CMakeLists.txt

```diff
@@ -84,8 +84,16 @@ if(CMAKE_COMPILER_IS_CLANG)
     set(CMAKE_C_FLAGS_RELEASE     "-O2")
     set(CMAKE_C_FLAGS_DEBUG       "-O0 -g3")
     set(CMAKE_C_FLAGS_COVERAGE    "-O0 -g3 --coverage")
-    set(CMAKE_C_FLAGS_ASAN        "-Werror -fsanitize=address -fno-common -fsanitize=undefined -fno-sanitize-recover -O3")
-    set(CMAKE_C_FLAGS_ASANDBG     "-Werror -fsanitize=address -fno-common -fsanitize=undefined -fno-sanitize-recover -O1 -g3 -fno-omit-frame-pointer -fno-optimize-sibling-calls ")
+    EXECUTE_PROCESS( COMMAND ${CMAKE_C_COMPILER} --version OUTPUT_VARIABLE clang_full_version_string )
+    string (REGEX REPLACE ".*clang version ([0-9]+\\.[0-9]+).*" "\\1" CLANG_VERSION_STRING ${clang_full_version_string})
+
+    if(CLANG_VERSION_STRING VERSION_GREATER 3.7)
```

Comment:
Did a quick test with clang version 3.6 and 3.5. ```=undefined,integer``` works with 3.6 but fails with 3.5. Hence it could be ```> 3.5``` or ```>= 3.6```.

Reply:
I also tested this and saw the same, but my thought was to change only what is needed, as I ma not entirely sure what's the difference, and couldn't find the documentation